lib/service/service.go

```diff
@@ -2570,6 +2571,21 @@ func (process *TeleportProcess) initAuthService() error {
 		return trace.Wrap(err)
 	})
 
+	accessMonitoringService, err := accessmonitoring.NewAccessMonitoringService(accessmonitoring.Config{
+		Logger: logger.With(
+			teleport.ComponentKey,
+			teleport.Component(teleport.ComponentAuth, "access_monitoring_service")),
+		Backend: b,
+		Client:  authServer,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	process.RegisterFunc("auth.access_monitoring_service", func() error {
```

Comment:
What's going to happen for hosted plugins? We currently run hosted plugins in teleport.e's plugin manager, and this looks a lot like a plugin but runs as an auth service. I'd like to understand the goal: what should run where, and what's the criteria. My main concern is the increasing number of different ways of running things in teleport.

Reply:
This service should be functionally the same as a hosted plugin for access reviews. We we're originally thinking about implementing it as a hosted plugin, but it made more sense to implement it as an auth service. The hosted plugins will continue to be managed by teleport.e's plugin manager.

I'm not sure if it'll work out as I'm planning, but once we're done with the initial implementation of automatic reviews, I'd like to reimplement the hosted plugins using this new `AccessMonitor`. This will replace the `App` that we use for plugins. And hopefully, we can eventually unify the different implementations we have of the `App`.